src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -89,25 +91,63 @@ public void setClient(LanguageClient client) {
         this.client = Optional.of(client);
     }
 
+    public Optional<File> getRoot() {
+        return Optional.ofNullable(project).map(proj -> proj.getRoot());
+    }
+
     /**
      * Processes extensions.
      * <p>
      * 1. Downloads external dependencies as jars 2. Creates a model from just
      * external jars 3. Updates locations index with symbols found in external jars
      *
-     * @param ext extensions
+     * @param ext  extensions
+     * @param root workspace root
      */
     public void createProject(SmithyBuildExtensions ext, File root) {
         Either<Exception, SmithyProject> loaded = SmithyProject.load(ext, root);
         if (loaded.isRight()) {
             this.project = loaded.getRight();
+            clearAllDiagnostics();
             sendInfo("Project loaded with " + this.project.getExternalJars().size() + " external jars and "
                     + this.project.getSmithyFiles().size() + " discovered smithy files");
         } else {
             sendError("Failed to create Smithy project: " + loaded.getLeft().toString());
         }
     }
 
+    /**
+     * Discovers Smithy build files and loads the smithy project defined by them.
+     *
+     * @param root workspace root
+     */
+    public void createProject(File root) {
```

Comment:
Moved from language server, because we need to have access to several fully configured settings, and it's easier this way.

If it becomes annoying here, it should probably be a static on `SmithyProject`